operate/services/manage.py

```diff
@@ -748,9 +750,10 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
                 # TODO This has to be removed for Optimus (needs to be properly implemented). Needs to be put back for Trader!
                 or current_agent_bond != staking_params["min_staking_deposit"]
                 or on_chain_description != service.description
+                or (current_staking_program is None)
+                ^ (user_params.staking_program_id == NO_STAKING_PROGRAM_ID)
```

Comment:
Very difficult to understand what you mean with a XOR here. Can you elaborate?

Reply:
here it means that if it was no_staking currently and want to stake or is already staked and want to run in no_staking. Then it should update